dotcom-rendering/src/web/components/SecureSignupIframe.importable.tsx

```diff
@@ -0,0 +1,368 @@
+		);
+
+		// The response body could be accessed with await response.text()
+		// here and added to state but the response is not informative
+		// enough to convey the actualreason for a failure to the user,
+		// so a generic failure message is used.
+		setIsWaitingForResponse(false);
+		setResponseOk(response.ok);
+
+		sendTracking(
+			newsletterId,
+			response.ok
+				? 'success-response-received'
+				: 'fail-response-received',
+		);
+	};
+
+	const resetForm: ReactEventHandler<HTMLButtonElement> = () => {
+		setErrorMessage(undefined);
+		setResponseOk(undefined);
+		recaptchaRef.current?.reset();
+	};
+
+	const handleCaptchaLoadError: ReactEventHandler<HTMLDivElement> = () => {
+		sendTracking(newsletterId, 'captcha-load-error');
+		setErrorMessage(`Sorry, the reCAPTCHA failed to load.`);
+		recaptchaRef.current?.reset();
+	};
+
+	const handleCaptchaComplete = (token: string | null) => {
+		if (!token) {
+			sendTracking(newsletterId, 'captcha-not-passed');
+			return;
+		}
+		sendTracking(newsletterId, 'captcha-passed');
+		setIsWaitingForResponse(true);
+		submitForm(token).catch((error) => {
+			// eslint-disable-next-line no-console -- unexpected error
+			console.error(error);
+			sendTracking(newsletterId, 'form-submit-error');
+			setErrorMessage(`Sorry, there was an error signing you up.`);
+			setIsWaitingForResponse(false);
+		});
+	};
+
+	const resizeIframe = (requestedHeight = 0): void => {
+		const { current: iframe } = iframeRef;
+		if (!iframe) {
+			return;
+		}
+		// verifiying the body is present before accessing the scrollHeight is necessary
+		// iframe.contentDocument?.body.scrollHeight can cause a TypeError
+		// the typing assumes body is always present on a Document but the use of
+		// srcDoc seems to allow the document to exist without the body.
+
+		const body = iframe.contentDocument?.body;
+		const scrollHeight = body ? body.scrollHeight : 0;
+		setIFrameHeight(Math.max(0, requestedHeight, scrollHeight + 15));
+	};
+
+	const resetIframeHeight = (): void => {
+		resizeIframe();
+	};
+
+	const handleClickInIFrame = (): void => {
+		sendTracking(newsletterId, 'click-button');
+	};
+
+	const handleSubmitInIFrame = (event: Event): void => {
+		event.preventDefault();
+		if (isWaitingForResponse) {
+			return;
+		}
+		setErrorMessage(undefined);
+		sendTracking(newsletterId, 'open-captcha');
+		recaptchaRef.current?.execute();
+	};
+
+	const attachListenersToIframe = () => {
+		const { current: iframe } = iframeRef;
+		iframe?.contentWindow?.addEventListener('resize', resetIframeHeight);
+		const form = iframe?.contentDocument?.querySelector('form');
+		const button = iframe?.contentDocument?.querySelector('button');
+		button?.addEventListener('click', handleClickInIFrame);
+		form?.addEventListener('submit', handleSubmitInIFrame);
+		resetIframeHeight();
+	};
+
+	// read siteKey
+	useEffect(() => {
+		setCaptchaSiteKey(
+			window.guardian.config.page.googleRecaptchaSiteKey ?? '',
```

Comment:
Does this need to be in a useeffect, or could you so something like 

`const captchaSiteKey = typeof window !== 'undefined' ?? window.guardian.config.page.googleRecaptchaSiteKey`

Reply:
You're right, this doesn't need to be in useEffect. 
I see a lot of other importable components set an `isServer` variable using `typeof window` so I'll adapt that pattern.